lib/aptible/cli/helpers/operation.rb

```diff
@@ -48,13 +48,52 @@ def cancel_operation(operation)
           operation.update!(cancelled: true)
         end
 
+        def operation_logs(operation)
+          res = get_operation_logs_redirect(operation)
+          # note: res :location is the  header "location" for a 301
+          s3_file_request = get_operation_logs_s3_file(res[:location])
+
+          m = "Printing out results of operation logs for #{operation.id}"
+          CLI.logger.info m
+          puts s3_file_request.body
+        end
+
         def prettify_operation(o)
           bits = [o.status, o.type, "##{o.id}"]
           if o.resource.respond_to?(:handle)
             bits.concat ['on', o.resource.handle]
           end
           bits.join ' '
         end
+
+        private
+
+        def get_operation_logs_redirect(operation)
+          uri = URI(operation.logs_url)
+          headers = { 'Authorization' => "Bearer #{fetch_token}" }
+          http = Net::HTTP.new(uri.host, uri.port)
```

Comment:
I think we need to add `require 'net/http'` to `lib/aptible/cli/agent.rb`.

Reply:
More context: I was getting this error when running my tests before:

```
/usr/lib/ruby/gems/2.4.0/bundler/gems/aptible-cli-c208a1917050/lib/aptible/cli/helpers/operation.rb:74:in `get_operation_logs_redirect': uninitialized constant Aptible::CLI::Helpers::Operation::Net (NameError)
Did you mean?  Set
```

It was fixed when I made the change mentioned above. Unfortunately, it still claims it can't get the operation logs despite the fact that the API is sending the expected response.

```
Requesting operation logs for succeeded deploy #2524 on test-app-2137b9c3-ba7d-48ec-b9c0-847a47a815ba...
Unable to retrieve operation logs with 301.
```

Reply:
Clarifying comment - josh ran into this when running integration tests locally.

i will pull this `net/http` in and see if these tests pass at that point

Reply:
Ok the reason it was still reporting that it wasn't getting the logs is that the Location header is under `'location'` (string) not `:location` (symbol) 🙄. This must have to do with the version of the `net/http` module.

Reply:
wow, so even requiring that won't let us proceed because integration tests are run in a slightly older version of ruby? not only where net/http isn't imported but also such an import won't save us because string vs symbol changes in headers referencing the Location has changed?

why do we support multiple versions of ruby for the CLI when we distribute it as a binary again (in CI/testing)?

Reply:
it looks like we build on 2.3 at the moment ( i validated on both mac and debian ). i followed a suggestion to use `.try` for these flavors for the time being

Reply:
I ended up making some other changes afterwards and actually tested it on both versions of ruby (<2.4 and 2.6.x in unit tests)

<img width="527" alt="image" src="https://user-images.githubusercontent.com/2961973/188901504-baf54da5-0c12-4f67-bf99-0053828b4ff9.png">

<img width="561" alt="image" src="https://user-images.githubusercontent.com/2961973/188904511-343dd03d-4130-44b3-b5f3-e66c90c1233c.png">